aloha-core/src/test/scala/com/eharmony/aloha/dataset/vw/multilabel/VwMultilabelRowCreatorTest.scala

```diff
@@ -0,0 +1,228 @@
+    val tests = Seq(
+      //         positive labels:
+      1,             // {}
+      6,             // {6}
+      7,             // {7}
+      8,             // {8}
+      6 * 7,         // {6, 7}
+      6 * 8,         // {6, 8}
+      7 * 8,         // {7, 8}
+      6 * 7 * 8      // {6, 7, 8}
+    )
+
+    // Could throw, but if so, test should fail.
+    val rc = getRowCreator(allLabelInts)
+
+    tests foreach { test =>
+      val exp = positiveAndNegativeValues(test, allLabelInts)
+      val (_, act) = rc(test)
+      testOutput(test, exp, act, MultilabelOutputPrefix, rc.classNs)
+    }
+  }
+
+  private[this] def getRowCreator(allLabelInts: Vector[Int]) = {
+    val rcProducer =
+      new VwMultilabelRowCreator.Producer[Int, Label](allLabelInts.map(y => y.toString))
+
+    val rcTry = rcProducer.getRowCreator(
+      CompiledSemanticsInstances.anyNameIdentitySemantics[Int],
+      MultilabelDatasetJson
+    )
+
+    // Don't care about calling `.get`.  If it fails, the test will appropriately blow up.
+    rcTry.get
+  }
+
+  private[this] def positiveAndNegativeValues(n: Int, allLabels: Seq[Int]): Seq[Boolean] = {
+    val divisorsN = divisors(n)
+    val labelInd: Map[Int, Int] = allLabels.zipWithIndex.toMap
+    val allLabelSet = labelInd.keySet
+    val pos = allLabelSet intersect divisorsN
+    val neg = allLabelSet diff divisorsN
+
+    // Because pos and neg form a partition of allLabelSet, labelInd.apply is safe.
+    val posAndNeg =
+      pos.map(p => labelInd(p) -> true) ++
+      neg.map(p => labelInd(p) -> false)
+
+    posAndNeg
+      .toSeq
+      .sorted
+      .map { case (_, isPos) => isPos }
+  }
+
+  private[this] def testOutput(
+      n: Int,
+      expectedResults: Seq[Boolean],
+      actualResults: Array[String],
+      prefix: Seq[String],
+      labelNs: Char
+  ): Unit = {
+
+    val suffix =
+      expectedResults.zipWithIndex map { case (isPos, i) =>
+        s"$i:${if (isPos) -1 else 0} |$labelNs _$i"
+      }
+
+    assertEquals(prefix, actualResults.take(prefix.size).toSeq)
+    assertEquals(suffix, actualResults.drop(prefix.size).toSeq)
+  }
+
+
+  /**
+    * Get all divisors of `n`.
+    * This is not super efficient: O(sqrt(N))
+    * @param n value for which divisors are desired.
+    * @return
+    */
+  private[this] def divisors(n: Int): Set[Int] =
+    (1 to math.sqrt(n).ceil.toInt).flatMap {
+      case i if n % i == 0 => List(i, n / i)
+      case _ => Nil
+    }(breakOut)
+}
+
+object VwMultilabelRowCreatorTest {
+  private type Domain = Map[String, Any]
+  private type Label = String
+  private val Omitted = ""
+  private val LabelsInTrainingSet = Vector("zero", "one", "two")
+  private val NegDummyClass = Int.MaxValue.toLong + 1
+  private val PosDummyClass = NegDummyClass + 1
+  private val PosVal = -1
+  private val NegVal = 0
+  private val X = Map.empty[String, Any]
+  private val SharedPrefix = "shared "
+
+  private val DummyLabels = List(
+    s"$NegDummyClass:$NegVal |y N",
```

Comment:
I would encourage you to not write `N` and `P` here, but to use the constants you created in the main code.